tests/Repositories/TaskRepositoryTest.php

```diff
@@ -99,6 +103,115 @@ public function it_can_retrieve_task_list_of_given_projects()
+    }
+
+    /** @test */
+    public function test_can_return_index_one_when_task_does_not_have_task_number()
+    {
+        $project = factory(Project::class)->create();
+        factory(Task::class)->create([
+            'project_id'  => $project->id,
+            'task_number' => '',
+        ]);
+        $getUniqueIndex = $this->taskRepo->getIndex($project->id);
+
+        $this->assertNotEmpty($getUniqueIndex);
+        $this->assertEquals(1, $getUniqueIndex);
+    }
+
+    /** @test */
+    public function test_can_get_my_tasks_of_given_project_id()
+    {
+        $user = factory(User::class)->create();
+        $task = factory(Task::class)->create();
+        $task->taskAssignee()->sync([$user->id]);
+
+        $loggedInUserTask1 = factory(Task::class)->create();
+        $loggedInUserTask1->taskAssignee()->sync([$this->defaultUserId]);
+
+        $loggedInUserTask2 = factory(Task::class)->create(['status' => Task::STATUS_COMPLETED]);
+        $loggedInUserTask2->taskAssignee()->sync([$this->defaultUserId]);
+
+        $getMyTasks = $this->taskRepo->myTasks(['project_id' => $loggedInUserTask1->project_id]);
+
+        $this->assertCount(5, $getMyTasks['activities']);
+
+        $this->assertCount(1, $getMyTasks['tasks']);
+        $this->assertEquals($loggedInUserTask1->id, $getMyTasks['tasks'][0]->id);
+        $this->assertEquals($loggedInUserTask1->title, $getMyTasks['tasks'][0]->title);
+        $this->assertNotEquals(Task::STATUS_COMPLETED, $getMyTasks['tasks'][0]->status);
+        $this->assertEquals($this->defaultUserId, $loggedInUserTask1->fresh()->taskAssignee[0]->id);
+    }
+
+    /** @test */
+    public function test_can_get_my_tasks()
+    {
+        $user = factory(User::class)->create();
+        $task = factory(Task::class)->create();
+        $task->taskAssignee()->sync([$user->id]);
+
+        $loggedInUserTask1 = factory(Task::class)->create();
+        $loggedInUserTask1->taskAssignee()->sync([$this->defaultUserId]);
+
+        $loggedInUserTask2 = factory(Task::class)->create(['status' => Task::STATUS_COMPLETED]);
+        $loggedInUserTask2->taskAssignee()->sync([$this->defaultUserId]);
+
+        $getMyTasks = $this->taskRepo->myTasks();
+
+        $this->assertCount(1, $getMyTasks['tasks']);
+        $this->assertEquals($loggedInUserTask1->id, $getMyTasks['tasks'][0]->id);
+        $this->assertEquals($loggedInUserTask1->title, $getMyTasks['tasks'][0]->title);
+    }
+
+    /** @test */
+    public function test_can_get_task_details_with_task_duration()
+    {
+        $timeEntry = factory(TimeEntry::class)->create(['duration' => 5]);
+        factory(TimeEntry::class)->create();
+
+        $getMyTask = $this->taskRepo->getTaskDetails($timeEntry->task_id);
+
+        $this->assertEquals($timeEntry->task_id, $getMyTask->id);
+        $this->assertEquals('00 Hours and 05 Minutes', $getMyTask->totalDuration);
+    }
+
+    /** @test */
+    public function test_can_get_task_details_when_duration_is_zero()
```

Comment:
this not need to cover, as we are now not allowed to add time entry with duration zero